[2.0] Fix M48 for delta printers/ABL compile issue

cdedwards opened the pull request:
fix M48 for delta printers. home printer if not already homed.

cdedwards committed:
fix M48 for delta printers

fix M48 for delta printers. home printer if not already homed.

Roxy-3D commented:
I don't understand why this fixes M48 for Delta's.     This homes the printer for all machine types.

cdedwards commented:
the if ENABLED(DELTA) fixes the delta. otherwise with the DISABLED(DELTA) it will not compile at all.

cdedwards committed:
remove ENABLED(DELTA)

remove ENABLED(DELTA) as it is not needed.

thinkyhead committed:
Update M48.cpp

cdedwards committed:
abl needs queue.h

without queue.h abl will not compile since there is a script that runs at the end of it and we need to queue it up.

thinkyhead commented:
The specification for `M48` simply requires homing first. The `axis_unhomed_error` function prints a message to the console when not homed. In future it may give audio feedback as well.
